### ISSUE TITLE
Maybe I don't know what I am doing

### ISSUE THREAD

**Issue by rw86347** [OlafenwaMoses/ImageAI#115]
I found two very similar bottles of alcohol.  I then tied a string to their necks and then took video of them from all sides.  This produced something like 3000 images of each of the two bottles.  I then trained until the accuracy hit 100%.  (epoch 25).  I then wrote a python script that would use the webcam and held the bottles in front of the camera.  I *think* the predictions were right 25% of the time.  But more concerning was when I held nothing in front of the camera. In that case it would still guess 90% chance it was one of the bottles of alcohol.  

Why won't it recognize nothing as low probability for both bottles?

Is there some trick I can do for super fine accuracy?

from imageai.Prediction.Custom import ModelTraining

model_trainer = ModelTraining()
model_trainer.setModelTypeAsResNet()
model_trainer.setDataDirectory("products")

model_trainer.trainModel(num_objects=2, num_experiments=100, enhance_data=True, batch_size=12, show_network_summary=True)

**Comment by jameswellsiv** [comment]
I think you might need to include images where there are no bottles so the classifier can learn what it would look like if the object wasn't in the image

**Comment by OlafenwaMoses** [comment]
> I think you might need to include images where there are no bottles so the classifier can learn what it would look like if the object wasn't in the image

Exactly... The model was to recognize 2 bottle types, which means it assumes everything else it sees falls into these 2 categories.

**Comment by OlafenwaMoses** [comment]
And considering the bottles are very similar, and you took their pictures of the same objects from different sides, your **3000** images most likely amount to just **50/100** unique view of the bottles which led to your model overfitting. **Overfitting** was probably why your model achieved 100% accuracy but failed in test. You will need more truly diverse images of the bottles.
